src/compiler/array.rs

```diff
@@ -61,6 +61,39 @@ pub(crate) fn data_ptr(symbol: &Symbol) -> Symbol {
     }
 }
 
+pub(crate) fn compile_at(compiler: &mut Compiler, array: &Symbol, index: &Symbol) -> Symbol {
+    let result = compiler.memory.allocate_symbol(array.type_.clone());
+    compiler
+        .memory
+        .read(compiler.instructions, array.memory_addr + 1, 2);
+    compiler
+        .memory
+        .read(compiler.instructions, index.memory_addr, 2);
+    // [index, len, data]
+    compiler.instructions.extend([
+        Instruction::Swap,
+        Instruction::Dup(Some(1)),
+        // [index, len, index, data]
+        Instruction::U32CheckedLT,
+        Instruction::Assert,
+        Instruction::Push(array.type_.miden_width()),
+        // [element_width, len, data_ptr]
+        Instruction::U32CheckedMul,
+        // [len * element_width, data_ptr]
+        Instruction::U32CheckedAdd,
+        // [data_ptr + len * element_width]
+        Instruction::MemLoad(None),
```

Comment:
This will only work for elements with width = 1. I made this for a task I'm working on right now:

```rust
pub(crate) fn get(compiler: &mut Compiler, arr: &Symbol, index: &Symbol) -> Symbol {
    assert!(matches!(arr.type_, Type::Array(_)));
    assert!(matches!(
        index.type_,
        Type::PrimitiveType(PrimitiveType::UInt32)
    ));

    let result = compiler
        .memory
        .allocate_symbol(element_type(&arr.type_).clone());

    compiler.instructions.extend([
        Instruction::MemLoad(Some(data_ptr(&arr).memory_addr)),
        // [data_ptr]
        Instruction::MemLoad(Some(index.memory_addr)),
        // [index, data_ptr]
        Instruction::Push(element_type(&arr.type_).miden_width()),
        // [element_width, index, data_ptr]
        Instruction::U32CheckedMul,
        // [offset = index * element_width, data_ptr]
        Instruction::U32CheckedAdd,
        // [ptr = data_ptr + offset]
    ]);

    for i in 0..element_type(&arr.type_).miden_width() {
        compiler.instructions.extend([
            Instruction::Dup(None),
            // [ptr, ptr]
            Instruction::Push(i),
            // [i, ptr, ptr]
            Instruction::U32CheckedAdd,
            // [ptr + i, ptr]
            Instruction::MemLoad(None),
            // [value, ptr]
            Instruction::MemStore(Some(result.memory_addr + i)),
            // [ptr]
        ]);
    }

    compiler.instructions.push(Instruction::Drop);

    result
}
```